lib/features/mailbox_dashboard/presentation/controller/search_controller.dart

```diff
@@ -0,0 +1,194 @@
+import 'dart:async';
+
+import 'package:core/core.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:jmap_dart_client/jmap/core/sort/comparator.dart';
+import 'package:jmap_dart_client/jmap/core/unsigned_int.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_comparator.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_comparator_property.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/base/base_controller.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/state/quick_search_email_state.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/usecases/quick_search_email_interactor.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/mailbox_dashboard_controller.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/search/email_receive_time_type.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/search/quick_search_filter.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/model/search/search_email_filter.dart';
+import 'package:tmail_ui_user/features/thread/domain/constants/thread_constants.dart';
+import 'package:tmail_ui_user/features/thread/domain/model/search_query.dart';
+import 'package:tmail_ui_user/features/thread/presentation/model/search_state.dart';
+import 'package:tmail_ui_user/features/thread/presentation/model/search_status.dart';
+
+class SearchController extends BaseController {
+  final QuickSearchEmailInteractor _quickSearchEmailInteractor;
+
+  final _mailboxDashBoardController = Get.find<MailboxDashBoardController>();
+
+  final searchInputController = TextEditingController();
+  final searchEmailFilter = SearchEmailFilter().obs;
+  final searchState = SearchState.initial().obs;
+  final isAdvancedSearchViewOpen = false.obs;
+  final dateFilterSelectedFormAdvancedSearch = EmailReceiveTimeType.allTime.obs;
+  final mailboxFilterSelectedFormAdvancedSearch = Rxn<PresentationMailbox>();
+  final hasAttachment = false.obs;
+
+  get _accountId =>  _mailboxDashBoardController.accountId;
+  get _userProfile =>  _mailboxDashBoardController.userProfile;
+
+  FocusNode searchFocus = FocusNode();
+
+  SearchEmailFilter get searchEmailFilterValue => searchEmailFilter.value;
+
+  SearchController(
+      this._quickSearchEmailInteractor,
+      );
+
+  selectOpenAdvanceSearch() {
+    isAdvancedSearchViewOpen.value = !isAdvancedSearchViewOpen.value;
```

Comment:
you should using  function `toggle()` of `GetX` for object `boolean obx`
Ex: `isAdvancedSearchViewOpen.toggle();`